dvc/commands/data.py

```diff
@@ -0,0 +1,178 @@
+            items = stage_status
+            if isinstance(stage_status, dict):
+                items = {
+                    file: state
+                    for state, files in stage_status.items()
+                    for file in files
+                }
+            if not items:
+                continue
+            yield stage, items
+
+    @classmethod
+    def _show_status(cls, status: "DataStatus") -> int:
+        git_info = status.pop("git")  # type: ignore[misc]
+        result = dict(cls._process_status(status))
+        if not result:
+            no_changes = "No changes"
+            if git_info.get("is_empty", False):
+                no_changes += " in an empty git repo"
+            ui.write(f"{no_changes}.")
+
+        for idx, (stage, stage_status) in enumerate(result.items()):
+            if idx:
+                ui.write()
+
+            label = cls.LABELS.get(stage, stage.capitalize() + " files")
+            header = f"{label}:"
+            color = cls.COLORS.get(stage, "normal")
+
+            ui.write(header)
+            if hint := cls.HINTS.get(stage):
+                ui.write(f"  ({hint})")
+
+            if isinstance(stage_status, dict):
+                items = [
+                    ": ".join([state, file])
+                    for file, state in stage_status.items()
+                ]
+            else:
+                items = stage_status
+
+            for item in items:
+                ui.write(f"\t[{color}]{item}[/]".expandtabs(8), styled=True)
+
+        if (hint := cls.HINTS.get("git_dirty")) and git_info.get("is_dirty"):
+            message = hint.format("other " if result else "")
+            ui.write(f"[blue]({message})[/]", styled=True)
+        return 0
+
+    def run(self) -> int:
+        with log_durations(logger.trace, "in data_status"):  # type: ignore
+            status = self.repo.data_status(
+                granular=self.args.granular,
+                untracked_files=self.args.untracked_files,
+                with_dirs=self.args.with_dirs,
+            )
+
+        if not self.args.unchanged:
+            status.pop("unchanged")  # type: ignore[misc]
+        if self.args.untracked_files == "no":
+            status.pop("untracked")
+        if self.args.json:
+            status.pop("git")  # type: ignore[misc]
+            ui.write_json(compact(status))
+            return 0
+        return self._show_status(status)
+
+
+def add_parser(subparsers, parent_parser):
+    data_parser = subparsers.add_parser(
+        "data",
+        parents=[parent_parser],
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    data_subparsers = data_parser.add_subparsers(
+        dest="cmd",
+        help="Use `dvc data CMD --help` to display command-specific help.",
+    )
+    fix_subparsers(data_subparsers)
+
+    DATA_STATUS_HELP = (
+        "Show changes between the last git commit, "
+        "the dvcfiles and the workspace."
```

Comment:
Oh and migrating discussion from https://github.com/iterative/dvc.org/pull/3812#pullrequestreview-1059748858 cc @skshetry @shcheklein:

```suggestion
    DATA_STATUS_HELP = (
        "Show changes between the last Git commit, "
        "DVC metafiles, and the workspace."
```

Reply:
> metafile is only docs-concept
this term has not been introduced in DVC (maybe for good reasons).

I think the only reason is that we don't have a system to ensure consistency in terminology between docs and help output but we probably should. It would improve the UX.

> dvcfiles is also a weird term
we should... not introduce internal terms into help / docs.

➕

And it sounds like `.dvc` files specifically. What about `dvc.lock`?

Reply:
This could be further simplified though, to just:

`Show changes in DVC-tracked data between the last Git commit and the workspace.`

Reply:
Metafile is a confusing term. Tbh I don't even know what it is, I can guess that it's a file having metadata about something. Also, it's a logical concept, not a physical one like `.dvc` and `dvc.yaml`, so I find it to be an unnecessary redirection. Unlike docs, we can afford repetition which is not that many.

Here, I think we can just avoid mentioning the files at all.
> Show changes in the data tracked by DVC in the workspace.

Reply:
> Show changes in DVC-tracked data between the last Git commit and the workspace.

yep, we can avoid metafiles, dvcfiles.

-----

my 2cs - I find both terms suboptimal, but would probably prefer metafiles - since, yes they are technically metafiles - they contain a spec for data, yes metadata about data. At least it sounds reasonable to me. Plus we already use it in docs. May be we can avoid using this low lever terminology in help messages at all btw- that would the best option for end users.

Reply:
> Here, I think we can just avoid mentioning the files at all.
we can avoid using this low lever terminology in help messages

➕➕